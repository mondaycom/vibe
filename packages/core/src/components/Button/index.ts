export { default as Button, type ButtonProps } from "./Button";

export * from "./Button.types";
