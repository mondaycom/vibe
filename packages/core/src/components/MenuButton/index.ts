export { default as MenuButton, type MenuButtonProps } from "./MenuButton";

export * from "./MenuButton.types";
