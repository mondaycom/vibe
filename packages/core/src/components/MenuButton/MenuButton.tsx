import React, { forwardRef, useCallback, useMemo, useRef, useState } from "react";
import cx from "classnames";
import { camelCase } from "lodash-es";
import { isForwardRef } from "react-is";
import Dialog, { DialogEvent } from "../Dialog/Dialog";
import DialogContentContainer from "../DialogContentContainer/DialogContentContainer";
import Tooltip, { TooltipProps } from "../Tooltip/Tooltip";
import useIsomorphicLayoutEffect from "../../hooks/ssr/useIsomorphicLayoutEffect";
import useMergeRef from "../../hooks/useMergeRef";
import { BUTTON_ICON_SIZE, SMALL_BUTTON_ICON_SIZE } from "../Button/ButtonConstants";
import { ElementContent, VibeComponent, VibeComponentProps, withStaticProps } from "../../types";
import {
  MenuButtonComponentPosition as MenuButtonComponentPositionEnum,
  MenuButtonSize as MenuButtonSizeEnum
} from "./MenuButtonConstants";
import {
  DialogPosition as DialogPositionEnum,
  HideShowEvent as DialogTriggerEventEnum,
  DialogSize as DialogSizeEnum
} from "../Dialog/DialogConstants";
import { MenuButtonComponentPosition, MenuButtonSize } from "./MenuButton.types";
import { DialogOffset, DialogPosition, DialogSize, DialogTriggerEvent } from "../Dialog/Dialog.types";
import { NOOP } from "../../utils/function-utils";
import { Menu } from "@vibe/icons";
import { getStyle } from "../../helpers/typesciptCssModulesHelper";
import { ComponentDefaultTestId, getTestId } from "../../tests/test-ids-utils";
import { MenuChild } from "../Menu/Menu/MenuConstants";
import styles from "./MenuButton.module.scss";
import { TooltipPositions } from "../Tooltip/Tooltip.types";
import { ComponentVibeId } from "../../tests/constants";

const MOVE_BY = { main: 8, secondary: 0 };

export interface MenuButtonProps extends VibeComponentProps {
  /**
   * Control the button's selected state
   */
  active?: boolean;
  /**
   *Class name to add to the button when the dialog is open
   */
  openDialogComponentClassName?: string;
  /**
   * Receives React Component
   */
  component?: (() => JSX.Element) | React.ElementType;
  size?: MenuButtonSize;
  open?: boolean;
  onClick?: (event: React.MouseEvent) => void;
  zIndex?: number;
  ariaLabel?: string;
  /*
    Class name to provide the element which wraps the popover/modal/dialog
   */
  dialogClassName?: string;
  /**
   * main - `dialogOffset.main` - main axis offset; `dialogOffset.secondary` secondary axis offset
   */
  dialogOffset?: DialogOffset;
  dialogPaddingSize?: DialogSize;
  dialogPosition?: DialogPosition;
  dialogShowTriggerIgnoreClass?: string;
  dialogHideTriggerIgnoreClass?: string;
  /**
   * the container selector in which to append the dialog
   * for examples: "body" , ".my-class", "#my-id"
   */
  dialogContainerSelector?: string;
  /**
   * Dialog Alignment
   */
  startingEdge?: string;
  /*
    Callback function to be called when the menu is shown
   */
  onMenuShow?: () => void;
  /*
  Callback function to be called when the menu is hidden
 */
  onMenuHide?: () => void;
  /**
   * Text to be displayed after the icon
   */
  text?: string;
  disabled?: boolean;
  tooltipContent?: string;
  /**
   Remove "Tab" key from the hide trigger
   */
  removeTabCloseTrigger?: boolean;
  /**
   is an array with the content of types:
   CLICK, CLICK_OUTSIDE, ESCAPE_KEY, TAB_KEY, MOUSE_ENTER, MOUSE_LEAVE,
   ENTER, MOUSE_DOWN, FOCUS, BLUR, CONTENT_CLICK
   */
  tooltipTriggers?: DialogTriggerEvent | DialogTriggerEvent[];
  /**
   * the disabled/tooltip position of the menu button - one of the MenuButton.dialogPositions
   */
  tooltipPosition?: TooltipPositions;
  /**
   * Tooltip Element Wrapper ClassName
   */
  tooltipReferenceClassName?: string;
  tooltipProps?: Partial<TooltipProps>;
  /**
   * When the MenuButton is hidden hide the dialog and tooltip as well
   */
  hideWhenReferenceHidden?: boolean;
  children?: ElementContent;
  /**
   * Specifies whether to render the component before or after the text
   */
  componentPosition?: MenuButtonComponentPosition;
  /**
   * Element to be used as the trigger element for the Menu - default is button
   */
  triggerElement?: React.ElementType;
  /**
   * Close the menu when an item is clicked
   */
  closeMenuOnItemClick?: boolean;
  /**
   * Whether tooltip should appear only when the trigger element is hovered and not the menu dialog
   */
  showTooltipOnlyOnTriggerElement?: boolean;
  closeDialogOnContentClick?: boolean;
}

const MenuButton: VibeComponent<MenuButtonProps> & {
  sizes?: typeof MenuButtonSizeEnum;
  paddingSizes?: typeof DialogContentContainer.sizes;
  dialogPositions?: typeof DialogPositionEnum;
  hideTriggers?: typeof Dialog.hideShowTriggers;
  componentPositions?: typeof MenuButtonComponentPositionEnum;
} = forwardRef(
  (
    {
      id,
      className,
      openDialogComponentClassName,
      children,
      component = Menu,
      componentPosition = "start",
      size = "small",
      open = false,
      onClick = NOOP,
      zIndex = null,
      ariaLabel = "Menu",
      closeMenuOnItemClick,
      dialogOffset = MOVE_BY,
      dialogPosition = Dialog.positions.BOTTOM_START,
      dialogClassName,
      dialogPaddingSize = DialogContentContainer.sizes.SMALL,
      dialogShowTriggerIgnoreClass,
      dialogHideTriggerIgnoreClass,
      onMenuHide = NOOP,
      onMenuShow = NOOP,
      disabled = false,
      text,
      tooltipContent,
      tooltipProps,
      tooltipTriggers = [MenuButton.hideTriggers.MOUSE_LEAVE],
      tooltipPosition = "right",
      startingEdge = "bottom",
      removeTabCloseTrigger = false,
      tooltipReferenceClassName,
      hideWhenReferenceHidden = true,
      dialogContainerSelector,
      active,
      triggerElement: TriggerElement = "button",
      showTooltipOnlyOnTriggerElement,
      "data-testid": dataTestId,
      closeDialogOnContentClick = false
    }: MenuButtonProps,
    ref
  ) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRef(ref, componentRef);

    const [isOpen, setIsOpen] = useState(open);
    const isActive = active ?? isOpen;

    const handleMenuClose = useCallback(
      (focusOnMenuButtonAfterClose: boolean) => {
        onMenuHide();
        setIsOpen(false);
        const button = componentRef.current;
        if (!button || !focusOnMenuButtonAfterClose) {
          return;
        }
        window.requestAnimationFrame(() => {
          button.focus();
        });
      },
      [onMenuHide]
    );

    const onMenuDidClose = useCallback(
      (event: React.KeyboardEvent) => {
        // TODO: check the functionality of the isEscapeKey since the event is not an actual KeyboardEVent but an object with propagate property only
        const isEscapeKey = event?.key === "Escape";
        if (isEscapeKey || closeMenuOnItemClick) {
          // @ts-ignore
          if (event.propagate) {
            handleMenuClose(isEscapeKey);
          }
        }
      },
      [closeMenuOnItemClick, handleMenuClose]
    );

    const onDialogDidHide = useCallback(
      (event: DialogEvent, hideEvent: string) => {
        handleMenuClose(hideEvent === Dialog.hideShowTriggers.ESCAPE_KEY);
      },
      [handleMenuClose]
    );

    const onDialogDidShow = useCallback(() => {
      setIsOpen(true);
      onMenuShow();
    }, [setIsOpen, onMenuShow]);

    const [clonedChildren, hideTrigger] = useMemo(() => {
      const triggers = new Set([
        Dialog.hideShowTriggers.CLICK_OUTSIDE,
        Dialog.hideShowTriggers.TAB_KEY,
        Dialog.hideShowTriggers.ESCAPE_KEY
      ]);

      if (closeDialogOnContentClick) {
        triggers.add(Dialog.hideShowTriggers.CONTENT_CLICK);
        triggers.add(Dialog.hideShowTriggers.ENTER);
      }

      if (removeTabCloseTrigger) {
        triggers.delete(Dialog.hideShowTriggers.TAB_KEY);
      }
      const childrenArr = React.Children.toArray(children) as MenuChild[];
      const cloned = childrenArr.map(child => {
        if (!React.isValidElement(child)) return null;

        const newProps: { focusOnMount?: boolean; onClose?: (event: React.KeyboardEvent) => void } = {};
        if (child.type && child.type.supportFocusOnMount) {
          newProps.focusOnMount = true;
          triggers.delete(Dialog.hideShowTriggers.ESCAPE_KEY);
        }

        if (child.type && child.type.isMenu) {
          newProps.onClose = onMenuDidClose;
        }

        return React.cloneElement(child, newProps);
      });
      return [cloned, Array.from(triggers)];
    }, [children, onMenuDidClose, closeDialogOnContentClick, removeTabCloseTrigger]);

    const content = useMemo(() => {
      if (clonedChildren.length === 0) return null;
      return (
        <DialogContentContainer size={dialogPaddingSize} type={DialogContentContainer.types.POPOVER}>
          {clonedChildren}
        </DialogContentContainer>
      );
    }, [clonedChildren, dialogPaddingSize]);

    const computedDialogOffset = useMemo(
      () => ({
        ...MOVE_BY,
        ...dialogOffset
      }),
      [dialogOffset]
    );

    const onMouseUp = (event: React.MouseEvent<HTMLElement>) => {
      if (disabled) {
        event.currentTarget.blur();
        return;
      }
      onClick(event);
    };

    const Icon = component;
    const iconSize = useMemo(() => {
      switch (size) {
        case "xxs":
        case "xs":
          return SMALL_BUTTON_ICON_SIZE;
        case "small":
        case "medium":
        case "large":
          return BUTTON_ICON_SIZE;
        default:
          return 24;
      }
    }, [size]);
    const icon = Icon ? <Icon size={iconSize.toString()} role="img" aria-hidden="true" /> : null;

    useIsomorphicLayoutEffect(() => {
      setIsOpen(open);
    }, [open, setIsOpen]);

    // Trigger element props, which are only relevant for "button" element, but might be needed for other elements e.g. Button
    const triggerElementProps =
      TriggerElement === "button"
        ? {
            ref: mergedRef
          }
        : {
            active: isActive,
            disabled: disabled,
            ref: isForwardRef(TriggerElement) ? mergedRef : undefined
          };

    const triggerElementNode = (
      <TriggerElement
        id={id}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.MENU_BUTTON, id)}
        data-vibe={ComponentVibeId.MENU_BUTTON}
        type="button"
        className={cx(styles.wrapper, className, getStyle(styles, camelCase(`size-${size}`)), {
          [styles.active]: isActive,
          [getStyle(styles, openDialogComponentClassName)]: isOpen && openDialogComponentClassName,
          [styles.disabled]: disabled,
          [styles.text]: text
        })}
        aria-haspopup="true"
        aria-expanded={isOpen}
        aria-label={!text && ariaLabel}
        onMouseUp={onMouseUp}
        aria-disabled={disabled}
        {...triggerElementProps}
      >
        {componentPosition === "start" && icon}
        {text && <span className={styles.innerText}>{text}</span>}
        {componentPosition === "end" && icon}
      </TriggerElement>
    );

    const dialogNode = (dialogChildren: React.ReactElement) => (
      <Dialog
        wrapperClassName={dialogClassName}
        position={dialogPosition}
        containerSelector={dialogContainerSelector}
        startingEdge={startingEdge}
        animationType="expand"
        content={content}
        moveBy={computedDialogOffset}
        showTrigger={disabled ? [] : ["click", "enter"]}
        hideTrigger={hideTrigger}
        showTriggerIgnoreClass={dialogShowTriggerIgnoreClass}
        hideTriggerIgnoreClass={dialogHideTriggerIgnoreClass}
        useDerivedStateFromProps={true}
        onDialogDidShow={onDialogDidShow}
        onDialogDidHide={onDialogDidHide}
        zIndex={zIndex}
        isOpen={isOpen}
        hideWhenReferenceHidden={hideWhenReferenceHidden}
      >
        {dialogChildren}
      </Dialog>
    );

    const tooltipNode = (tooltipChildren: React.ReactElement) => (
      <Tooltip
        content={tooltipContent}
        position={tooltipPosition}
        showTrigger="mouseenter"
        hideTrigger={tooltipTriggers}
        referenceWrapperClassName={tooltipReferenceClassName}
        hideWhenReferenceHidden={hideWhenReferenceHidden}
        {...tooltipProps}
      >
        {tooltipChildren}
      </Tooltip>
    );

    if (showTooltipOnlyOnTriggerElement) {
      return dialogNode(tooltipNode(triggerElementNode));
    }
    return tooltipNode(dialogNode(triggerElementNode));
  }
);

export default withStaticProps(MenuButton, {
  sizes: MenuButtonSizeEnum,
  paddingSizes: DialogSizeEnum,
  dialogPositions: DialogPositionEnum,
  hideTriggers: DialogTriggerEventEnum,
  componentPositions: MenuButtonComponentPositionEnum
});
