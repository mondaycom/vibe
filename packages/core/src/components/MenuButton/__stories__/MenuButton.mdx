import { Meta } from "@storybook/blocks";
import {
  BUTTON,
  ICON_BUTTON,
  SPLIT_BUTTON
} from "../../../storybook/components/related-components/component-description-map";
import * as MenuButtonStories from "./MenuButton.stories";

<Meta of={MenuButtonStories} />

# MenuButton

MenuButton is a component that opens a Dialog next to a button, the content of the dialog could be anything you want.

<Canvas of={MenuButtonStories.Overview} />

### Import

```js
import { MenuButton } from "@vibe/core";
```

## Props

<PropsTable />

## Usage

<UsageGuidelines
  guidelines={[
    "When you want to place content next to an element",
    "When the content needs to be on top of adjacent content "
  ]}
/>

## Variants

### Sizes

<Canvas of={MenuButtonStories.Sizes} />

### Different Icon

<Canvas of={MenuButtonStories.DifferentIcon} />

### With Text

<Canvas of={MenuButtonStories.WithText} />

### With Text and Icon at the end

<Canvas of={MenuButtonStories.WithTextAndIconAtTheEnd} />

### Disabled

<Canvas of={MenuButtonStories.Disabled} />

### Custom trigger element

You can use any element as a trigger for the menu button, just pass it as a <code>triggerElement</code> to the <code>MenuButton</code> component.

<Canvas of={MenuButtonStories.CustomTriggerElement} />

## Related components

<RelatedComponents componentsNames={[BUTTON, ICON_BUTTON, SPLIT_BUTTON]} />
