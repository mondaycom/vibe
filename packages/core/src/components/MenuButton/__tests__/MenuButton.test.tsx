import { describe, it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render, screen, waitFor, waitForElementToBeRemoved } from "@testing-library/react";
import MenuButton from "../MenuButton";
import { Bolt } from "@vibe/icons";
import Button from "../../Button/Button";
import MenuItem from "../../Menu/MenuItem/MenuItem";
import Menu from "../../Menu/Menu/Menu";
import { userEvent } from "@storybook/testing-library";

describe("MenuButton", () => {
  it("renders correctly with empty props", () => {
    const tree = renderer
      .create(
        <MenuButton>
          <div>Menu</div>
        </MenuButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with size Large", () => {
    const tree = renderer
      .create(
        <MenuButton size="large">
          <div>Menu</div>
        </MenuButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with Bolt Icon", () => {
    const tree = renderer
      .create(
        <MenuButton component={Bolt} text="Hello">
          <div>Menu</div>
        </MenuButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with .dummy-class-name", () => {
    const tree = renderer
      .create(
        <MenuButton className="dummy-class-name">
          <div>Menu</div>
        </MenuButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with tooltip content", () => {
    const tree = renderer
      .create(
        <MenuButton tooltipContent="tooltip content">
          <div>Menu</div>
        </MenuButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with showTooltipOnlyOnTriggerElement and tooltip content", () => {
    const tree = renderer
      .create(
        <MenuButton showTooltipOnlyOnTriggerElement tooltipContent="tooltip content">
          <div>Menu</div>
        </MenuButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with showTooltipOnlyOnTriggerElement and without tooltip content", () => {
    const tree = renderer
      .create(
        <MenuButton showTooltipOnlyOnTriggerElement>
          <div>Menu</div>
        </MenuButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with a default Menu icon at the end", () => {
    const tree = renderer
      .create(
        <MenuButton text="Hello" componentPosition="start">
          <div>Menu</div>
        </MenuButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with Button triggerElement", () => {
    const tree = renderer
      .create(
        <MenuButton
          triggerElement={props => (
            <Button kind="primary" {...props} className={""}>
              Button
            </Button>
          )}
        ></MenuButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO - fix this test
  it.skip("renders correctly with open state", () => {
    const tree = renderer
      .create(
        <MenuButton open>
          <div>Menu</div>
        </MenuButton>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("Click", () => {
    it("should show menu", async () => {
      const { getByLabelText } = render(
        <MenuButton ariaLabel="Menu Button">
          <div>Menu</div>
        </MenuButton>
      );
      const button = getByLabelText("Menu Button");
      await waitFor(() => {
        fireEvent.click(button);
      });
      const menu = await screen.findByText("Menu");
      expect(menu).toBeTruthy();
    });
  });

  describe("Item Click", () => {
    it.each`
      closeMenuOnItemClick | expected
      ${true}              | ${"close"}
      ${false}             | ${"not close"}
    `(
      "should $expected menu after clicking enter when closeMenuOnItemClick is $closeMenuOnItemClick",
      async ({ closeMenuOnItemClick }) => {
        const { getByLabelText } = render(
          <MenuButton ariaLabel="Menu Button" closeMenuOnItemClick={closeMenuOnItemClick}>
            <Menu>
              <MenuItem title="Menu Item" />
            </Menu>
          </MenuButton>
        );
        const button = getByLabelText("Menu Button");
        await waitFor(async () => {
          fireEvent.click(button);
          expect(await screen.findAllByText("Menu Item")).toBeTruthy();
          userEvent.type(button, "{enter}");
        });
        if (closeMenuOnItemClick) {
          await waitForElementToBeRemoved(() => screen.queryByText("Menu Item"));
          expect(screen.queryByText("Menu Item")).toBeFalsy();
        } else {
          expect(screen.queryByText("Menu Item")).toBeTruthy();
        }
      }
    );
  });
});
