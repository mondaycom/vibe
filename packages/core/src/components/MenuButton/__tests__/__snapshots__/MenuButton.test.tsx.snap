// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton renders correctly with .dummy-class-name 1`] = `
<button
  aria-disabled={false}
  aria-expanded={false}
  aria-haspopup="true"
  aria-label="Menu"
  className="wrapper dummy-class-name sizeSmall"
  data-testid="menu-button"
  data-vibe="MenuButton"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  <svg
    aria-hidden="true"
    fill="currentColor"
    height="20"
    role="img"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      d="M6 10.5C6 11.3284 5.32843 12 4.5 12 3.67157 12 3 11.3284 3 10.5 3 9.67157 3.67157 9 4.5 9 5.32843 9 6 9.67157 6 10.5zM11.8333 10.5C11.8333 11.3284 11.1618 12 10.3333 12 9.50492 12 8.83334 11.3284 8.83334 10.5 8.83334 9.67157 9.50492 9 10.3333 9 11.1618 9 11.8333 9.67157 11.8333 10.5zM17.6667 10.5C17.6667 11.3284 16.9951 12 16.1667 12 15.3383 12 14.6667 11.3284 14.6667 10.5 14.6667 9.67157 15.3383 9 16.1667 9 16.9951 9 17.6667 9.67157 17.6667 10.5z"
      fill="currentColor"
    />
  </svg>
</button>
`;

exports[`MenuButton renders correctly with Bolt Icon 1`] = `
<button
  aria-disabled={false}
  aria-expanded={false}
  aria-haspopup="true"
  aria-label={false}
  className="wrapper sizeSmall text"
  data-testid="menu-button"
  data-vibe="MenuButton"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  <svg
    aria-hidden="true"
    fill="currentColor"
    height="20"
    role="img"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      clipRule="evenodd"
      d="M11.3336 2.28375C11.6473 2.38119 11.8611 2.67147 11.8611 2.99999V7.80555H15C15.2785 7.80555 15.5341 7.95989 15.6638 8.20639C15.7934 8.45289 15.7758 8.75094 15.618 8.98045L9.50692 17.8693C9.32081 18.1401 8.98017 18.2581 8.66643 18.1607C8.35269 18.0632 8.13889 17.773 8.13889 17.4444V12.6389H5C4.72148 12.6389 4.4659 12.4845 4.33624 12.238C4.20657 11.9915 4.22418 11.6935 4.38197 11.464L10.4931 2.5751C10.6792 2.30438 11.0198 2.1863 11.3336 2.28375ZM6.42577 11.1389H8.88889C9.3031 11.1389 9.63889 11.4747 9.63889 11.8889V15.0297L13.5742 9.30555H11.1111C10.6969 9.30555 10.3611 8.96976 10.3611 8.55555V5.41475L6.42577 11.1389Z"
      fill="currentColor"
      fillRule="evenodd"
    />
  </svg>
  <span
    className="innerText"
  >
    Hello
  </span>
</button>
`;

exports[`MenuButton renders correctly with Button triggerElement 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorPrimary"
  data-testid="menu-button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`MenuButton renders correctly with a default Menu icon at the end 1`] = `
<button
  aria-disabled={false}
  aria-expanded={false}
  aria-haspopup="true"
  aria-label={false}
  className="wrapper sizeSmall text"
  data-testid="menu-button"
  data-vibe="MenuButton"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  <svg
    aria-hidden="true"
    fill="currentColor"
    height="20"
    role="img"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      d="M6 10.5C6 11.3284 5.32843 12 4.5 12 3.67157 12 3 11.3284 3 10.5 3 9.67157 3.67157 9 4.5 9 5.32843 9 6 9.67157 6 10.5zM11.8333 10.5C11.8333 11.3284 11.1618 12 10.3333 12 9.50492 12 8.83334 11.3284 8.83334 10.5 8.83334 9.67157 9.50492 9 10.3333 9 11.1618 9 11.8333 9.67157 11.8333 10.5zM17.6667 10.5C17.6667 11.3284 16.9951 12 16.1667 12 15.3383 12 14.6667 11.3284 14.6667 10.5 14.6667 9.67157 15.3383 9 16.1667 9 16.9951 9 17.6667 9.67157 17.6667 10.5z"
      fill="currentColor"
    />
  </svg>
  <span
    className="innerText"
  >
    Hello
  </span>
</button>
`;

exports[`MenuButton renders correctly with empty props 1`] = `
<button
  aria-disabled={false}
  aria-expanded={false}
  aria-haspopup="true"
  aria-label="Menu"
  className="wrapper sizeSmall"
  data-testid="menu-button"
  data-vibe="MenuButton"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  <svg
    aria-hidden="true"
    fill="currentColor"
    height="20"
    role="img"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      d="M6 10.5C6 11.3284 5.32843 12 4.5 12 3.67157 12 3 11.3284 3 10.5 3 9.67157 3.67157 9 4.5 9 5.32843 9 6 9.67157 6 10.5zM11.8333 10.5C11.8333 11.3284 11.1618 12 10.3333 12 9.50492 12 8.83334 11.3284 8.83334 10.5 8.83334 9.67157 9.50492 9 10.3333 9 11.1618 9 11.8333 9.67157 11.8333 10.5zM17.6667 10.5C17.6667 11.3284 16.9951 12 16.1667 12 15.3383 12 14.6667 11.3284 14.6667 10.5 14.6667 9.67157 15.3383 9 16.1667 9 16.9951 9 17.6667 9.67157 17.6667 10.5z"
      fill="currentColor"
    />
  </svg>
</button>
`;

exports[`MenuButton renders correctly with showTooltipOnlyOnTriggerElement and tooltip content 1`] = `
<span
  className=""
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button
    aria-disabled={false}
    aria-expanded={false}
    aria-haspopup="true"
    aria-label="Menu"
    className="wrapper sizeSmall"
    data-testid="menu-button"
    data-vibe="MenuButton"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="20"
      role="img"
      viewBox="0 0 20 20"
      width="20"
    >
      <path
        d="M6 10.5C6 11.3284 5.32843 12 4.5 12 3.67157 12 3 11.3284 3 10.5 3 9.67157 3.67157 9 4.5 9 5.32843 9 6 9.67157 6 10.5zM11.8333 10.5C11.8333 11.3284 11.1618 12 10.3333 12 9.50492 12 8.83334 11.3284 8.83334 10.5 8.83334 9.67157 9.50492 9 10.3333 9 11.1618 9 11.8333 9.67157 11.8333 10.5zM17.6667 10.5C17.6667 11.3284 16.9951 12 16.1667 12 15.3383 12 14.6667 11.3284 14.6667 10.5 14.6667 9.67157 15.3383 9 16.1667 9 16.9951 9 17.6667 9.67157 17.6667 10.5z"
        fill="currentColor"
      />
    </svg>
  </button>
</span>
`;

exports[`MenuButton renders correctly with showTooltipOnlyOnTriggerElement and without tooltip content 1`] = `
<span
  className=""
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button
    aria-disabled={false}
    aria-expanded={false}
    aria-haspopup="true"
    aria-label="Menu"
    className="wrapper sizeSmall"
    data-testid="menu-button"
    data-vibe="MenuButton"
    onMouseUp={[Function]}
    type="button"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="20"
      role="img"
      viewBox="0 0 20 20"
      width="20"
    >
      <path
        d="M6 10.5C6 11.3284 5.32843 12 4.5 12 3.67157 12 3 11.3284 3 10.5 3 9.67157 3.67157 9 4.5 9 5.32843 9 6 9.67157 6 10.5zM11.8333 10.5C11.8333 11.3284 11.1618 12 10.3333 12 9.50492 12 8.83334 11.3284 8.83334 10.5 8.83334 9.67157 9.50492 9 10.3333 9 11.1618 9 11.8333 9.67157 11.8333 10.5zM17.6667 10.5C17.6667 11.3284 16.9951 12 16.1667 12 15.3383 12 14.6667 11.3284 14.6667 10.5 14.6667 9.67157 15.3383 9 16.1667 9 16.9951 9 17.6667 9.67157 17.6667 10.5z"
        fill="currentColor"
      />
    </svg>
  </button>
</span>
`;

exports[`MenuButton renders correctly with size Large 1`] = `
<button
  aria-disabled={false}
  aria-expanded={false}
  aria-haspopup="true"
  aria-label="Menu"
  className="wrapper sizeLarge"
  data-testid="menu-button"
  data-vibe="MenuButton"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  <svg
    aria-hidden="true"
    fill="currentColor"
    height="20"
    role="img"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      d="M6 10.5C6 11.3284 5.32843 12 4.5 12 3.67157 12 3 11.3284 3 10.5 3 9.67157 3.67157 9 4.5 9 5.32843 9 6 9.67157 6 10.5zM11.8333 10.5C11.8333 11.3284 11.1618 12 10.3333 12 9.50492 12 8.83334 11.3284 8.83334 10.5 8.83334 9.67157 9.50492 9 10.3333 9 11.1618 9 11.8333 9.67157 11.8333 10.5zM17.6667 10.5C17.6667 11.3284 16.9951 12 16.1667 12 15.3383 12 14.6667 11.3284 14.6667 10.5 14.6667 9.67157 15.3383 9 16.1667 9 16.9951 9 17.6667 9.67157 17.6667 10.5z"
      fill="currentColor"
    />
  </svg>
</button>
`;

exports[`MenuButton renders correctly with tooltip content 1`] = `
<span
  className=""
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button
    aria-disabled={false}
    aria-expanded={false}
    aria-haspopup="true"
    aria-label="Menu"
    className="wrapper sizeSmall"
    data-testid="menu-button"
    data-vibe="MenuButton"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="20"
      role="img"
      viewBox="0 0 20 20"
      width="20"
    >
      <path
        d="M6 10.5C6 11.3284 5.32843 12 4.5 12 3.67157 12 3 11.3284 3 10.5 3 9.67157 3.67157 9 4.5 9 5.32843 9 6 9.67157 6 10.5zM11.8333 10.5C11.8333 11.3284 11.1618 12 10.3333 12 9.50492 12 8.83334 11.3284 8.83334 10.5 8.83334 9.67157 9.50492 9 10.3333 9 11.1618 9 11.8333 9.67157 11.8333 10.5zM17.6667 10.5C17.6667 11.3284 16.9951 12 16.1667 12 15.3383 12 14.6667 11.3284 14.6667 10.5 14.6667 9.67157 15.3383 9 16.1667 9 16.9951 9 17.6667 9.67157 17.6667 10.5z"
        fill="currentColor"
      />
    </svg>
  </button>
</span>
`;
